core/blockchain/blockchain_gateway.py

```diff
@@ -1,164 +1,154 @@
-import asyncio
 import logging
+from threading import Event, Timer
 import ipfsapi
 import time
-from typing import Callable
+from typing import Callable, Tuple
 
-from core.blockchain.blockchain_utils   import setter, getter, filter_diffs
+from core.blockchain.blockchain_utils   import filter_diffs, TxEnum
 from core.blockchain.blockchain_utils   import get_global_state, ipfs_to_content
-from core.utils.event_types             import RawEventTypes
-from core.blockchain.tx_utils           import TxEnum
+from core.utils.enums                   import RawEventTypes, MessageEventTypes
 
 
 logging.basicConfig(level=logging.DEBUG,
     format='[BlockchainGateway] %(message)s')
 
-
 class BlockchainGateway(object):
     """
     Blockchain Gateway 
 
     The blockchain gateway listens to the blockchain and notifies the appropriate classes
     inside the Unix Service when there is relevant information ready for them. Follows
-    an event-driven programming paradigm using a series of asyncio loops for listening.
-
-    # NOTE: For this PR we have left the functionality for the Developer in so that
-    we can actually test the Gateway. It will be left in until the Notebook can test.
+    an event-driven programming paradigm using a series of async loops for listening.
 
     In order for this to work, the following must be running:
         IPFS Daemon: `ipfs daemon`
-        The lotion app: `node app_trivial.js` from /Non-Trivial-Chain/on-chain-component
-
+        The lotion app: `node app_trivial.js` from dagora-chain
+    For more specific instructions check travis.yaml to see how travis does it.
     """
 
     def __init__(self):
         """
-        TODO: Refactor dependencies
-        TODO: deal with config
-        TODO: `communication_manager` is only used in a subset of methods,
-        consider separating
+        Initialize state to an empty list. Everything else is left to configure().
         """
         self.state = []
+        self.event = Event()
 
     def configure(self, config_manager: object, communication_manager: object):
```

Comment:
We're separating them for dependency injection and because it's not clear yet the order in which we will be bootstrapping modules in a way that they can all communicate between each other. This gives us flexibility.